### ISSUE TITLE
meteor update doesn`t update from 0.8.3. to 0.9.x (Win7)

### ISSUE THREAD

**Issue by mrvictorn** [sdarnell/meteor-win#54]
Tried to update, got this answer:
C:\dev\meteor\panorama>meteor update
This project is already at Meteor 0.8.3, the latest release.

**Comment by sdarnell** [comment]
That's right. I haven't completed the Windows port of Meteor 0.9.x yet, so 0.8.3 is the latest.

**Comment by aadamsx** [comment]
Meteor 0.9.x incorporated Meteorite (AFAIK).  From the Meteorite page, "meteorite's git based approach runs counter to the MSI installation that's required to get it working," will this become a problem porting Meteor 0.9.x to windows?  What issues are you running into while porting?  Just wondering as I cannot install a few packages because of Meteorite's incompatibility with Windows, and would like to move to Meteor 0.9.x to correct the package issues.

**Comment by sdarnell** [comment]
Meteor 0.9.x removes the need for Meteorite as the core package server aspects are included in the core.
There were a number of platform assumptions/dependencies in Meteorite and it was also unfortunately dependent on git. The new package system in 0.9.x is not dependent on git and instead serves packages directly over http. Unfortunately, the new package system chose to use a colon in file names and this is a no no for Windows, and I'm working through these issues and others...

However, having said all that, you do not need meteorite or the new package system to actually use packages on 0.8.3 ... even on Windows.  You just need to do a bit of downloading work manually, putting the packages in the right place.  I'm not saying it is trivial, but it is possible.

Basically clone iron-router and its dependencies at the right git hash, and put them in the app's packages directory, and then call `meteor add` for each package.
Have a look at: http://csharprambling.wordpress.com/2014/04/24/adding-meteor-package-in-windows/

**Comment by steph643** [comment]
Don't know if it helps, but I do find installing packages manually trivial. Here is what I do:
1- On the GitHub page of the package, follow the 'releases' link.
2- Download a release dated before August 2014 (hence supporting Meteor 0.8.3).
3- Check the package name by looking at the first line of the smart.json file of the package.
4- Unzip the release in the 'packages' folder of the project: projectName\packages\packageName.
4- Run 'meteor add packageName' or just add a packageName line in the .meteor/packages file.
5- Run Meteor: it will complain that it does not find such and such other packages (dependencies).
6- Apply steps 1 to 3 to the missing packages, until all dependencies are resolved.

**Comment by aadamsx** [comment]
Thanks for the feedback everyone.  I might give this work-around a try, but of course I'd much-prefer to wait for the next update if it's not too far away.  Might just break down and get a Macbook Air if the port process continues to be an impediment -- haha.

Thanks again for your work here Mr. Darnell.

**Comment by danielsvane** [comment]
Glad to know you're working on 0.9.x!

As for Meteorite support, there actually is a branch that works on Windows. I had to do some modifications to it, since it was very outdated. Try this guide, hopefully it still works: http://www.danielsvane.dk/blog/meteor-and-meteorite-on-windows (sorry for the self-promotion)

**Comment by steph643** [comment]
When starting with Meteor on Windows, I had searched for a Meteorite port, but found nothing. I think you should try to get it referenced here: http://win.meteor.com/

**Comment by danielsvane** [comment]
I expect @sdarnell is done with 0.9 soon enough, thus removing the need for Meteorite.

**Comment by vbezruchkin** [comment]
Hi guys,
+1 here for meteor .9x on windows. It looks like the guys at meteorjs added a lot of interesting features including phonegap / cordova support. 
Thanks in advance, @sdarnell

**Comment by darkadept** [comment]
Definitely a +1 for .9x on windows. I keep checking back here every other day to see if it's out yet! ;-) And yes, thanks @sdarnell!

**Comment by piffie** [comment]
Is there an update to this issue? Can we help with something?

**Comment by steph643** [comment]
+1 :-)

**Comment by sdarnell** [comment]
I'm really close now - hope to release a build today.
Thanks for your patience and support.

@vbezruchkin The first release will be 0.9.0.1 based so won't include phonegap/cordova support, that should follow, and hopefully there won't be too many issues there.

**Comment by vbezruchkin** [comment]
@sdarnell thanks for your work! Can't wait to put my hands on it. 

No problem with phonegap support, I tried it on a vagrant box and it seems to be a bit incomplete now so we can easily wait as much as needed.

**Comment by steph643** [comment]
He did it!

**Comment by piffie** [comment]
Yes! great!

2014-10-23 10:37 GMT+02:00 steph643 notifications@github.com:

**Comment by sdarnell** [comment]
I'm afraid it still doesn't auto-update from 0.8.3, but you can now download 0.9.0.1 from https://github.com/sdarnell/meteor/wiki/Windows

**Comment by danielsvane** [comment]
Thanks @sdarnell, really appreciate your work!

**Comment by peterennis** [comment]
Thanks for this.I tried the build and it failed, also noticed 0.8.3 was still the latest download.I will check with the new link first before trying to build again. 

```
 On Thursday, October 23, 2014 6:57 PM, Stephen Darnell <notifications@github.com> wrote:
```

 I'm afraid it still doesn't auto-update from 0.8.3, but you can now download 0.9.0.1 from https://github.com/sdarnell/meteor/wiki/Windows—

**Comment by sdarnell** [comment]
@peterennis please raise an issue with the details of how it failed.

As for 0.8.3 still showing up as the latest, this is deliberate. A. Because the auto update won't work (the changes in the package system are significant) B. Because the new system uses a different mechanism to detect newer releases - so updating the number would not help.

For now, and realistically forever, there isn't enough benefits of the auto upgrade given the technical issues (and effort involved). I'd rather spend the time getting Cordova working.

**Comment by sdarnell** [comment]
Just to check, you do need to use the LaunchMeteor09.exe not the previous LaunchMeteor.exe.

Once the build has had some exposure, I'll flip the switch and just have the LaunchMeteor.exe (installing a 0.9.x build)

**Comment by peterennis** [comment]
Sounds good.How about considering some "stable label" on the site rather thanjust LaunchMeteor.exeI find it clearer when the version number is part of the package nameand it is a more standard way of doing things e.g.http://nodejs.org/download/
Then provide info that points to GitHub for thoseconsidered beta testers. The can grab another versioned exe.
Alphas can grab the code (with suitable instructions),confirm the build process and potentially provide moresupport for the Cordova work.
I will raise an issue with my build failure and worktowards "suitable instructions".

```
 On Friday, October 24, 2014 11:46 AM, Stephen Darnell <notifications@github.com> wrote:
```

 Just to check, you do need to use the LaunchMeteor09.exe not the previous LaunchMeteor.exe.Once the build has had some exposure, I'll flip the switch and just have the LaunchMeteor.exe (installing a 0.9.x build)—

**Comment by sdarnell** [comment]
There is now a windows:METEOR@0.9.0.1-rc4, and you may need to reinstall as I suspect `meteor update --release windows:METEOR@0.9.0.1-rc4` may fail.

I'm unlikely to support automatic update from 0.8.3 to 0.9.x, given time constraints, and the problems it caused some users of the core meteor.
